### PR TITLE
feature: (added suggested user)

### PR DESCRIPTION
**Pull request by blikq**
# Pull Request: Add Suggested Users Feature

## Overview
This PR implements a comprehensive suggested users feature that allows users to discover and interact with user suggestions. The feature includes query functionality with pagination and the ability to dismiss suggestions.

### 1. Get Suggested Users
**Endpoint:** `GET /api/v1/suggested-users/`

**Description:** Retrieves a list of suggested users for the authenticated user with pagination support.

**Authentication:** Required (Bearer Token)

**Query Parameters:**
- `size` (integer, optional): Number of suggestions to return (max: 20, default: 10)
- `bookmark` (string, optional): Pagination bookmark for retrieving specific pages

**Request Example:**
```http
GET /api/v1/suggested-users/?size=10&bookmark=1
Authorization: Bearer <jwt_token>
```

**Response:**
```json
{
  "items": [
    {
      "id": 123,
      "nickname": "john_doe",
      "profile_image_url": "https://example.com/avatar.jpg",
      "profile_url": "https://example.com/users/john_doe",
      "description": "Software developer passionate about open source"
    }
  ],
  "total_count": 25
}
```

**Response Fields:**
- `items`: Array of suggested user summaries
  - `id`: Unique identifier for the suggestion
  - `nickname`: User's display name
  - `profile_image_url`: URL to user's profile image (optional)
  - `profile_url`: URL to user's profile page
  - `description`: User's bio/description (optional)
- `total_count`: Total number of available suggestions

### 2. Dismiss Suggested User
**Endpoint:** `POST /api/v1/suggested-users/{id}`

**Description:** Marks a specific user suggestion as dismissed, removing it from future suggestion lists.

**Authentication:** Required (Bearer Token)

**Path Parameters:**
- `id` (integer): The ID of the suggestion to dismiss

**Request Body:**
```json
{
  "dismissed": true
}
```

**Request Example:**
```http
POST /api/v1/suggested-users/123
Authorization: Bearer <jwt_token>
Content-Type: application/json

{
  "dismissed": true
}
```

**Response:**
```json
{}
```

### Testing Coverage
The implementation includes comprehensive test coverage:
- ✅ Basic query functionality
- ✅ Pagination behavior
- ✅ Size limit enforcement
- ✅ Authentication/authorization
- ✅ Dismiss functionality
- ✅ Data structure validation
- ✅ Edge cases (empty results, unauthorized access)
- ✅ Suggestion refresh and regeneration

## API Behavior Notes

1. **Suggestion Refresh:** Each query clears existing suggestions and generates new ones, ensuring users see fresh recommendations
2. **Size Limits:** The API enforces a maximum of 20 suggestions per request at the route level
3. **Authorization:** All endpoints require valid JWT authentication
4. **Idempotency:** Dismissing an already dismissed suggestion is safe and won't cause errors
5. **Data Isolation:** Users only see and can interact with their own suggestions

### CODE REVIEW

File: packages/dto/src/tables/users/user.rs [hunk 1]
```diff
@@ -43,6 +43,9 @@ pub struct User {
     #[api_model(many_to_many = group_members, foreign_table_name = groups, foreign_primary_key = group_id, foreign_reference_key = user_id)]
     #[serde(default)]
     pub groups: Vec<Group>,
+    #[api_model(one_to_many = suggested_users, foreign_primary_key = user_id)]
```
Comment (hackartists) [inline packages/dto/src/tables/users/user.rs:46]:
```suggestion
    #[api_model(one_to_many = suggested_users, foreign_key = user_id)]
```

one_to_may supports `foreign_key`
`many_to_many` supports `foreign_primary_key` and `foreign_reference_key` @blikq

**Comment by blikq** [comment]
@hackartists please review the updates have been made
